src/services/componentService.js

```diff
@@ -136,19 +143,26 @@ export default class ComponentService {
   async handleChangeEvent(event) {
     const details = event.detail;
     if (details) {
-      const { field, newValue } = details;
+      const { field, newValue, oldValue} = details;
       const component = event.target.tagName.toLowerCase();
-      const apiValue = ComponentService.convertValue(component, newValue);
+      const apiValue = ComponentService.convertValue(component, newValue, oldValue);
 
       event.target.setAttribute('loading', true);
 
       // Update post via API
       try {
-        await this.api.updatePost(this.postType, this.postId, {
+        const apiResponse= await this.api.updatePost(this.postType, this.postId, {
           [field]: apiValue,
         });
 
+        //Update value attribute in frontend
+        if(component==='dt-comm-channel'){
+          const valueAttri= apiResponse[field]
+          event.target.setAttribute('value',JSON.stringify(valueAttri))
+        }
+
```

Comment:
I looked at this a bit further, and I don't think this code is actually needed. I removed it and the value is updated properly. We should test this more, but I believe it can be removed.